s3/README.md

```diff
@@ -4,44 +4,58 @@ Module to create s3 bucket
 
 ## Input Variables
 
-- region
-  - string
-- profile
-  - string
-- bucket_name
-  - string
-- purpose
-  - string
-- owner
-  - string
-- team
-  - string
-- lazy_api_host
-  - string
-  - default = "https://lazy.apps.ops-drivevariant.com"
-- lazy_api_key
-  - string
-- octopus_project_space
-  - string
-- octopus_project_name
-  - string
+ | Name                         | Type          | Default             | Example           |    Notes           |
+ | ---------------------------- | ------------- | ------------------- | ----------------- | -----------------  |
+ | region                       | string        |                     |                   |                    |
+ | profile                      | string        |                     |                   |                    |
+ | bucket_name                  | string        |                     |                   |                    |
+ | lazy_api_host                | string        | "https://lazy.apps.ops-drivevariant.com"  |                    | auto set at octopus|
+ | lazy_api_key                 | string        |                     |                   |auto set at octopus |
+ | role_arn                     | string        |      ""             |                   |                    |
+ | user_tags                    | object        |                     |user_tags = {team = "devops4" purpose = "s3-test3" owner = "naveen3"}| For `user_tags`, refer <https://github.com/variant-inc/lazy-terraform/tree/master/submodules/tags>   |
+ | octopus_tags                 | object        |                     | octopus_tags = { project = "actions-test3" space = "Default3"}| auto set at octopus|
+ | replication                  | string        |  false              |                   |                    |
 
 ## Example .tfvars
 
 ```bash
 profile = "devops"
 region = "us-west-2"
-bucket_name = "naveen-ops-1"
-purpose = "devops"
-team = "devops"
-owner = "devops"
+bucket_name = "navin-ops-11"
 lazy_api_host = "https://lazy.apps.ops-drivevariant.com"
-lazy_api_key = "add_lazy_api_key_here"
-bucket = "lazy-tf-state20210107203535113800000001"
-key = "s3/default"
-dynamodb_table = "lazy_tf_state"
-s3_backend_region = "us-west-2"
-octopus_project_space = "test-space"
-octopus_project_name = "test-project"
-role_arn = "arn:aws:iam::123456789013:role/role_name"
+lazy_api_key = "####################"
+user_tags = {
+team = "devops4"
+purpose = "s3-test3"
+owner = "naveen3"
+}
+octopus_tags = {
+project = "actions-test3"
+space   = "Default3"
+}
+replication=true
+```
+
+## Example .tf file module reference
+
+```bash
+module "test_s3_module" {
+    source = "git::https://github.com/variant-inc/lazy-terraform.git//s3?ref=v1"
+        profile = "devops"
+        region = "us-west-2"
+        bucket_name = "navin-ops-11"
+        lazy_api_host = "https://lazy.apps.ops-drivevariant.com"
```

Comment:
Should the developers add these variables in all their modules?

Reply:
updated in read me on how to set these variables.